pkg/migrations/sql/idempotent/telemetry.sql

```diff
@@ -59,4 +59,73 @@ $$
+
+        SELECT approximate_row_count('SCHEMA_CATALOG.series')::TEXT INTO result;
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('series_total_approx', result);
+
+        SELECT n_distinct::TEXT INTO result FROM pg_stats WHERE schemaname='SCHEMA_TRACING' AND tablename='span' AND attname='trace_id';
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('traces_total_approx', result);
+
+        SELECT approximate_row_count('SCHEMA_TRACING.span')::TEXT INTO result;
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('spans_total_approx', result);
+
+        SELECT hypertable_size('SCHEMA_TRACING.span')::TEXT INTO result;
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('spans_bytes_total', result);
+
+        SELECT count(*)::TEXT INTO result FROM SCHEMA_CATALOG.label WHERE key = '__tenant__';
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('multi_tenancy_tenant_count', result);
+
+        SELECT count(*)::TEXT INTO result FROM timescaledb_information.data_nodes;
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('node_count', result);
+
+        SELECT count(*)::TEXT INTO result FROM SCHEMA_CATALOG.label_key WHERE key = '__cluster__';
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('ha_cluster_count', result);
+
+        SELECT count(*)::TEXT INTO result FROM SCHEMA_CATALOG.metric WHERE is_view IS true;
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('downsampling_count', result);
+
+        SELECT count(*)::TEXT INTO result FROM SCHEMA_CATALOG.exemplar;
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('exemplar_metrics_total', result);
+
+        SELECT count(*)::TEXT INTO result FROM SCHEMA_CATALOG.metadata;
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('metric_metadata_total', result);
+
+        SELECT value INTO result FROM SCHEMA_CATALOG.default WHERE key = 'retention_period';
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('default_metric_retention', result);
+
+        SELECT value INTO result FROM SCHEMA_CATALOG.default WHERE key = 'chunk_interval';
+        PERFORM SCHEMA_PS_CATALOG.apply_telemetry('default_chunk_interval', result);
+    END;
+$$
+LANGUAGE PLPGSQL;
+
+CREATE OR REPLACE FUNCTION SCHEMA_PS_CATALOG.apply_telemetry(telemetry_name TEXT, telemetry_value TEXT) RETURNS VOID AS
+$$
+    BEGIN
+        IF telemetry_value IS NULL THEN
+            telemetry_value := '0';
+        END IF;
+
+        -- First try to use promscale_extension to fill the metadata table.
+       SELECT update_tsprom_metadata(telemetry_name, telemetry_value, TRUE);
+
+        -- If promscale_extension is not installed, catch the exception and try the manual way.
+        EXCEPTION WHEN OTHERS THEN
+            INSERT INTO _timescaledb_catalog.metadata(key, value, include_in_telemetry) VALUES ('promscale_' || telemetry_name, telemetry_value, TRUE) ON CONFLICT (key) DO UPDATE SET value = EXCLUDED.value, include_in_telemetry = EXCLUDED.include_in_telemetry;
```

Comment:
```suggestion
            --if this fails, throw an error so that the connector can log (or not) as appropriate.
            INSERT INTO _timescaledb_catalog.metadata(key, value, include_in_telemetry) VALUES ('promscale_' || telemetry_name, telemetry_value, TRUE) ON CONFLICT (key) DO UPDATE SET value = EXCLUDED.value, include_in_telemetry = EXCLUDED.include_in_telemetry;
```